src/common/compress/helper.c

```diff
@@ -0,0 +1,278 @@
+            strPtr(compressHelperLocal[type].type));
+    }
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+/**********************************************************************************************************************************/
+const String *
+compressTypeStr(CompressType type)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(ENUM, type);
+    FUNCTION_TEST_END();
+
+    ASSERT(type < COMPRESS_LIST_SIZE);
+
+    FUNCTION_TEST_RETURN(compressHelperLocal[type].type);
+}
+
+/**********************************************************************************************************************************/
+CompressType
+compressTypeFromName(const String *name)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(STRING, name);
+    FUNCTION_TEST_END();
+
+    CompressType result = compressTypeNone + 1;
+
+    for (; result < COMPRESS_LIST_SIZE; result++)
+    {
+        if (strEndsWith(name, compressHelperLocal[result].ext))
+            break;
+    }
+
+    if (result == COMPRESS_LIST_SIZE)
+        result = compressTypeNone;
+
+    FUNCTION_TEST_RETURN(result);
+}
+
+/**********************************************************************************************************************************/
+int
+compressLevelDefault(CompressType type)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(ENUM, type);
+    FUNCTION_TEST_END();
+
+    ASSERT(type < COMPRESS_LIST_SIZE);
+    compressTypePresent(type);
+
+    FUNCTION_TEST_RETURN(compressHelperLocal[type].levelDefault);
+}
+
+/**********************************************************************************************************************************/
+IoFilter *
+compressFilter(CompressType type, int level)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(ENUM, type);
+        FUNCTION_TEST_PARAM(INT, level);
+    FUNCTION_TEST_END();
+
+    ASSERT(type < COMPRESS_LIST_SIZE);
+    ASSERT(type != compressTypeNone);
+    compressTypePresent(type);
+
+    FUNCTION_TEST_RETURN(compressHelperLocal[type].compressNew(level));
+}
+
+/**********************************************************************************************************************************/
+IoFilter *
+compressFilterVar(const String *filterType, const VariantList *filterParamList)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(STRING, filterType);
+        FUNCTION_LOG_PARAM(VARIANT_LIST, filterParamList);
+    FUNCTION_LOG_END();
+
+    ASSERT(filterType != NULL);
+
+    IoFilter *result = NULL;
+
+    for (CompressType compressType = compressTypeNone + 1; compressType < COMPRESS_LIST_SIZE; compressType++)
+    {
+        if (strBeginsWith(filterType, compressHelperLocal[compressType].type))
+        {
+            compressTypePresent(compressType);
+
+            switch (strPtr(filterType)[strSize(compressHelperLocal[compressType].type)])
+            {
+                case 'C':
```

Comment:
The question is where to put that documentation that anybody would find it?

Maybe we should just add the compress/decompress type names to compressHelperLocal and search for them explicitly?  This is something I'd like to improve with better interface functions in the future but at least this would make it clearer for now.

Reply:
yes, using the names would be good

Reply:
I see it has been resolved in this commit https://github.com/pgbackrest/pgbackrest/pull/950/commits/d6a12b24fda45219475f77f1b22098dbd58985af. Looks good.